packages/cli/src/commands/stream/deterministic.ts

```diff
@@ -0,0 +1,87 @@
+import { flags } from '@oclif/command'
+import { TileDocument } from '@ceramicnetwork/stream-tile'
+import type { TileMetadataArgs } from '@ceramicnetwork/stream-tile'
+
+import { Command } from '../../command'
+import type { CommandFlags } from '../../command'
+import { parseControllers } from '../../utils'
+
+type Flags = CommandFlags & {
+  controller?: string
+  metadata?: TileMetadataArgs
+}
+
+export default class Deterministic extends Command<
+  Flags,
+  {
+    content: string
+    schemaId?: string
+  }
+> {
+  static description = 'Create a new deterministic Stream'
+
+  static args = [
+    {
+      name: 'schema',
+      description: 'StreamID of desired Schema',
+      required: false,
+    },
+  ]
+
+  static flags = {
+    ...Command.flags,
+    controller: flags.string({
+      char: 'c',
+      description:
+        'Stream Controller, once set this is the only DID that will be able to update the stream.',
+    }),
+    metadata: flags.string({
+      char: 'm',
+      description: 'Stream Metadata',
+      parse: JSON.parse,
+    }),
+  }
+
+  async run(): Promise<void> {
+    this.spinner.start('Creating stream...')
+    let did: string | undefined
+    if (this.flags.key != null) {
+      did = this.authenticatedDID.id
+    } else {
+      throw new Error('No DID cached, please provide your key.')
+    }
+    try {
+      let parsedControllers: Array<string>
+      if (this.flags.controller !== undefined) {
+        parsedControllers = parseControllers(this.flags.controller)
+      } else if (did !== undefined) {
+        parsedControllers = parseControllers(did)
+      } else {
+        throw new Error('No DID to assign as a controller')
+      }
+
+      const metadata = {
+        controllers: this.flags.metadata?.controllers || parsedControllers,
+        tags: this.flags.metadata?.tags || [],
+        family: this.flags.metadata?.family || '',
+        schema: this.flags.metadata?.schema || '',
+        deterministic: true,
+        anchor: false,
+        publish: false,
+      }
+
+      const tile = await TileDocument.create(this.ceramic, metadata)
```

Comment:
That seems incorrect, `TileDocument.create()` takes the `content` as second argument. I think this should be:
```suggestion
      const tile = await TileDocument.deterministic(this.ceramic, this.args.metadata)
```

Reply:
Unfortunately, we can't use the `TileDocument.deterministic()` with the current version of TileDocument installed. I've attempted to upgrade it to support `TileDocument.deterministic()` however it creates a lot more issues than I can justify fitting into this request. I will be creating a story in ZenHub for this work to be followed up on as it is important. 

Currently I've gone ahead and done the following: 
```
      const tile = await TileDocument.create(this.ceramic, this.args.content, metadata)
```

Where within the metadata object I am forcibly setting it to deterministic with this command.

Reply:
If there are issues updating to a newer minor version of Ceramic that should be raised, what is the problem?
The Glaze packages are already using the new version of Ceramic so anyways better make sure the dependencies for the CLI are in sync before making any release.